src/components/course-settings/remove-teacher.cjsx

```diff
@@ -25,10 +24,11 @@ module.exports = React.createClass
       .value()
 
   goToDashboard: ->
-    @context.router.transitionTo('dashboard', {courseId: @props.courseId})
+    TeacherRosterStore.once 'deleted', ->
+      window.location.href = '/dashboard/'
```

Comment:
When using transitionTo I encountered a SecurityTransgression error b/c the logged in course role had been deleted.  The combination of using once 'deleted' and window.location bypasses any missing course role errors and the courselist is always updated when reaching /dashboard/